numpy/_core/fromnumeric.py

```diff
@@ -296,7 +304,27 @@ def reshape(a, newshape, order='C'):
            [3, 4],
            [5, 6]])
     """
-    return _wrapfunc(a, 'reshape', newshape, order=order)
+    if newshape is None and shape is None:
+        raise TypeError(
+            "reshape() missing 1 required positional argument: 'shape'")
+    if newshape is not None and shape is not None:
+        raise ValueError(
+            "You cannot specify 'newshape' and 'shape' arguments "
+            "at the same time.")
```

Comment:
I think this should be a `TypeError` (bad parameters usually are), not that it matters much.

Reply:
(might move it into the branch below, but doesn't really matter)

Reply:
Done!